service/history/api/updateworkflow/api.go

```diff
@@ -117,7 +117,8 @@ func (u *Updater) Invoke(
 	)
 
 	if err != nil {
-		return u.OnError(err)
+		u.OnError(err)
+		return nil, err
```

Comment:
You still want rejection response to be returned:
```suggestion
		rejResp := u.OnError(err)
		return rejResp, err
```

Reply:
Right!